pkg/alertmanager/amcfg.go

```diff
@@ -294,9 +317,18 @@ func (cg *configGenerator) convertReceiver(ctx context.Context, in *monitoringv1
 		}
 	}
 
+	var opsgenieConfigs []*opsgenieConfig
+	if l := len(in.OpsGenieConfigs); l > 0 {
+		opsgenieConfigs = make([]*opsgenieConfig, l)
+		for i := range in.OpsGenieConfigs {
+			opsgenieConfigs[i] = cg.convertOpsgenieConfig(ctx, in.OpsGenieConfigs[i], crKey)
```

Comment:
Let's wait for https://github.com/prometheus-operator/prometheus-operator/pull/3604 and after it gets merged reafctor this to handle incorrect opsgenie configs.

Reply:
I just rebased the master branch as this got merged